### PR TITLE
Dev krushna

### PR DESCRIPTION
**Pull request by mardikark-gslab**
This pull request contains below changes
- classify function for conventional way of KNN, updated run_knn interface
- NOT operator modification
- MIN & MAX operator modification to return index along with value
- INDEX operator implementation
- Minor modification in SUMBY & NUMBY
- Q.avg() implementation
- B1 write modification in vector

### CODE REVIEW

File: OPERATORS/F1S1OPF2/lua/vnot.tmpl [hunk 1]
```diff
@@ -36,12 +36,13 @@ ${fn}(
 
 {
   int status = 0;
```
Comment (subramon) [inline OPERATORS/F1S1OPF2/lua/vnot.tmpl]: put a comment as to why you are doing the ceil operator and what the potential pitfalls of that are (we are manipulating bits that we should not) but are doing that deliberately.

File: OPERATORS/GETK/lua/expander_getk.lua [hunk 2]
```diff
@@ -42,28 +52,30 @@ local function expander_getk(a, fval, k, optargs)
...
     end
-    -- copy from bufZ to bufX
-    local num_in_Z = ptr_num_in_Z[0]
-    ffi.C.memcpy(bufZ, bufX, num_in_Z * width)
-    nX = num_in_Z
     chunk_idx = chunk_idx + 1
   end
-  val_vec:put_chunk(buf1, n_buf1)
+  val_vec:put_chunk(bufX, nil, nX)
   return val_vec
```
Comment (subramon) [inline OPERATORS/GETK/lua/expander_getk.lua]: don;t we need an eov?

File: OPERATORS/GETK/lua/expander_getk.lua [hunk 3]
```diff
@@ -42,28 +52,30 @@ local function expander_getk(a, fval, k, optargs)
...
+      nX = num_in_Z
     end
-    -- copy from bufZ to bufX
-    local num_in_Z = ptr_num_in_Z[0]
-    ffi.C.memcpy(bufZ, bufX, num_in_Z * width)
-    nX = num_in_Z
     chunk_idx = chunk_idx + 1
   end
-  val_vec:put_chunk(buf1, n_buf1)
+  val_vec:put_chunk(bufX, nil, nX)
```
Comment (subramon) [inline OPERATORS/GETK/lua/expander_getk.lua]: don't we need an eov after this?

File: OPERATORS/GETK/lua/expander_getk.lua (hunk 2 above)
Comment (mardikark-gslab) [reply OPERATORS/GETK/lua/expander_getk.lua]: Updated the expander to have gen function, so eov() is performed underneath of eval().

**Pull request merged**